sections/earn/StakeGrid.tsx

```diff
@@ -18,14 +27,66 @@ const TimeRemainingData = () => {
 };
 
 const StakeGrid = () => {
+	const { t } = useTranslation();
 	const dispatch = useAppDispatch();
-	const earnedRewards = useAppSelector(({ earn }) => earn.earnedRewards);
+	const earnedRewards = useAppSelector(selectEarnedRewards);
+	const kwentaAmount = useAppSelector(selectKwentaAmount);
+	const wethAmount = useAppSelector(selectWethAmount);
+	const lpTotalSupply = useAppSelector(selectLPTotalSupply);
 	const yieldPerDay = useAppSelector(selectYieldPerDay);
+	const [tokenPrices, setTokenPrices] = useState<any>(null);
 
 	const handleClaim = useCallback(() => {
 		dispatch(claimRewards());
 	}, [dispatch]);
 
+	useEffect(() => {
+		const tokens = async () => {
+			const coinGeckoPrices = await sdk.exchange.batchGetCoingeckoPrices(
+				[KWENTA_COINGECKO_ADDRESS, ETH_COINGECKO_ADDRESS],
+				true
+			);
+
+			const kwentaPrice = coinGeckoPrices
+				? toWei(coinGeckoPrices[KWENTA_COINGECKO_ADDRESS]?.usd?.toString())
+				: zeroBN;
+
+			const wethPrice = coinGeckoPrices
+				? toWei(coinGeckoPrices[ETH_COINGECKO_ADDRESS]?.usd?.toString())
+				: zeroBN;
+
+			setTokenPrices({
+				kwentaPrice: kwentaPrice,
+				wethPrice: wethPrice,
+			});
+		};
+
+		(async () => {
+			await tokens();
+		})();
+	}, []);
+
+	const tvl = useMemo(
```

Comment:
I think we can move these memo'd values to `state/earn/selectors` as we'll aim to keep all data logic separate to UI components, which should help with performance as well.

Reply:
Thanks. Will make the changes.